Doc/library/os.rst

```diff
@@ -3780,6 +3780,168 @@ features:
    .. versionadded:: 3.10
 
 
+.. function:: timerfd_create(clockid, flags)
+
+   Create and return a timer file descriptor. The file descriptors supports
+   raw :func:`read` with a buffer size of 8, :func:`~select.select`, :func:`~select.poll`
+
+   .. seealso:: :manpage:`timerfd_create(2)`
+
+
+   Refer to :ref:`time-clock-id-constants` for a list of accepted values for *clockid*.
+
+   *flags* can be constructed from :const:`TFD_NONBLOCK` and :const:`TFD_CLOEXEC`.
+
+   If the timerfd counter is zero and :const:`TFD_NONBLOCK` is not
+   specified, :func:`read` blocks.
+
+   Example::
+
+       import time
+       import os
+
+       fd = os.timerfd_create(time.CLOCK_REALTIME, 0)
+
+       # timer interval 0.5 second
+       inteval = 0.5
+
+       # timer start in 1 second
+       value = 1
+
+       # start timer
+       _, _ = os.timerfd_settime(fd, 0, inteval, value)
+
+       try:
+           while:
+               # wait timerfd expired
+               _ = fd.read(8)
+               do_work()
+       finally:
+           os.close(fd)
+
+   Example::
+
+       import time
+       import os
+
+       fd = os.timerfd_create(time.CLOCK_REALTIME, 0)
+
+       # timer interval 0.5 second
+       inteval_ns = 10**9 // 2
+
+       # timer start in 1 second
+       value_ns = 10**9
+
+       # start timer in nano second unit
+       _, _ = os.timerfd_settime_ns(fd, 0, inteval_ns, value_ns)
+
+       try:
+           while:
+               # wait timerfd expired
+               _ = fd.read(8)
+               do_work()
+       finally:
+           os.close(fd)
+
+
+   .. availability:: Linux >= 2.6.27 with glibc >= 2.8
+
+   .. versionadded:: 3.13
+
+
+.. function:: timerfd_settime(fd, flags, interval, value)
+
+   Start/stop timer for a timer file descriptor.
+
+   .. seealso:: :manpage:`timerfd_settime(2)`
+
+   *interval* coresponds to :c:member:`!it_interval` in :c:struct:`itimerspec` and *value* coresponds to
+   :c:member:`!it_value` in :c:struct:`itimerspec`. They are both in seconds and types are double.
+
+   *interval* and *value* are interpreted like this to pass :manpage:`timerfd_settime(2)`.
+
+   .. c:var:: struct itimerspec new_value;
+   .. c:var:: double interval;
+   .. c:var:: double value;
+
+   :c:expr:`interval = new_value.it_interval.tv_sec + new_value.it_interval.tv_nsec * 1e-9`
+
+   :c:expr:`value = new_value.it_value.tv_sec + new_value.it_value.tv_nsec * 1e-9`
```

Comment:
This behaviour is surprising. I'd expect that using the nanosecond version of the function overrides any previous values, and vice-versa with the float version of the function -- but it seems their values are combined. I think users should be **warned** of this behaviour.

Reply:
@AA-Turner 
Added comment at 3d55e2ecff.

Reply:
I think it is much better to change the suprising behaviour if possible rather than adding a comment. However if you disagree with changing the behaviour, we should rephrase and make this a warning to the user.

```rest
.. caution::
   The seconds set by this function and the nanoseconds set by 
   :func:`timerfd_settime_ns` are summed by the timer descriptor, which may cause surpising results.
   For example, setting an interval of 2 seconds via this function and 2**9 seconds via :func:`timerfd_settime_ns` will make the actual interval of the timer file descriptor **4 seconds**, not 2.
```

Reply:
@AA-Turner 
What do you mean?
I'm not sure what is your concerns.

Setting an interval of 2 seconds via `timerfd_settime` will make the time file descriptor in every  2 seconds.

Setting an interval of 2**9 seconds(you mean 512 seconds?) via `timerfd_settime_ns` will make the time file descriptor in every 512 seconds.

And the original comment no longer exists on the lastest branch.